definitions/scenarios/upgrade_to_foreman_3_3.rb

```diff
@@ -0,0 +1,93 @@
+module Scenarios::Foreman_3_3
+  class Abstract < ForemanMaintain::Scenario
+    def self.upgrade_metadata(&block)
+      metadata do
+        tags :upgrade_scenario
+        confine do
+          feature(:foreman_install) &&
+            (feature(:foreman_server).current_minor_version == '3.2' || \
+            ForemanMaintain.upgrade_in_progress == '3.3')
+        end
+        instance_eval(&block)
+      end
+    end
+
+    def target_version
+      '3.3'
+    end
+  end
+
+  class PreUpgradeCheck < Abstract
+    upgrade_metadata do
+      description 'Checks before upgrading to Foreman 3.3'
+      tags :pre_upgrade_checks
+      run_strategy :fail_slow
+    end
+
+    def compose
+      add_steps(find_checks(:default))
+      add_steps(find_checks(:pre_upgrade))
+    end
+  end
+
+  class PreMigrations < Abstract
+    upgrade_metadata do
+      description 'Procedures before upgrading to Foreman 3.3'
+      tags :pre_migrations
+    end
+
+    def compose
+      add_steps(find_procedures(:pre_migrations))
+    end
+  end
+
+  class Migrations < Abstract
+    upgrade_metadata do
+      description 'Upgrade steps for Foreman 3.3'
+      tags :migrations
+      run_strategy :fail_fast
+    end
+
+    def set_context_mapping
+      context.map(:assumeyes, Procedures::Installer::Upgrade => :assumeyes)
+    end
+
+    def compose
+      add_step(Procedures::Repositories::Setup.new(:version => '3.3'))
+      if el?
+        modules_to_enable = ["foreman:#{el_short_name}"]
+        add_step(Procedures::Packages::EnableModules.new(:module_names => modules_to_enable,
+                                                         :assumeyes => true))
+      end
```

Comment:
Foreman 3.3 still supports EL7. Will we not package it for 3.2/3.3 on EL7 or is there a mechanism to make this only apply to EL 8? If so, why is the `if el?` even there?

Reply:
Apologies for confusion! The whole upstream upgrade story should be supporting from Foreman 3.4, I mean 3.4 to 3.5 upgrade. Considering we already have 3.4 near to GA then maybe from 3.5. However to test the upgrade scenarios I added the older upgrade paths.

Reply:
Ah, fair enough.